server/bin/pbench-audit-server.py

```diff
@@ -302,28 +303,17 @@ def verify_results(rhier, verifylist, user_arg):
             """
             continue
 
-        direct_entries = list_direct_entries(results_hierarchy, controller)
+        direct_entries = list_direct_entries(rhier.path, controller)
+        rhier.add_controller(controller)
 
-        if user_arg:
-            user_controller = f"{user_arg}/{controller}"
-            rhier.add_controller(user_controller)
-        else:
-            rhier.add_controller(controller)
-
-        for dirent in direct_entries:
-            path = Path(dirent.split(os.path.join(controller, ""), 1)[1])
-            dir_path = Path(dirent)
+        for dir_path in direct_entries:
+            path = Path(str(dir_path).split(os.path.join(controller, ""), 1)[1])
```

Comment:
This says to use `<controller>/` as the separator, split the `dir_path` there and take everything that's after the separator.  That's an indirect way of going about and I'm sure that Pathlib has better ways to deal with it. Please investigate and fix.

Reply:
I tried taking hold on it but pathlib doesn't seems to handle it.

Reply:
Add a comment: "Here, we may encounter an additional subdirectory structure because of "prefix" handling. The function `path_below_controller()' gives us the path of that subtructure."

Then write the function `path_below_controller()' to encapsulate the ugliness: that will simplify the reviewer's life (and that is not only other people: that may well be the future you :-)) , in that it will explain what it's doing without tripping them up. If somebody comes up with a better implementation in the future, they can change just that function, without disturbing the rest of the program.